power_grid_model_c/power_grid_model/include/power_grid_model/component/transformer.hpp

```diff
@@ -37,7 +37,9 @@ class Transformer : public Branch {
           winding_to_{transformer_input.winding_to},
           clock_{transformer_input.clock},
           tap_side_{transformer_input.tap_side},
-          tap_pos_{transformer_input.tap_pos},
+          tap_pos_{transformer_input.tap_pos == na_IntS
+                       ? (transformer_input.tap_nom == na_IntS ? (IntS)0 : transformer_input.tap_nom)
```

Comment:
How about we set it at `tap_nom` by default instead of `0`? That would disable the taps and make it safer.

Reply:
@nitbharambe it is set to `tap_nom`, but `tap_nom` is optional, so if `tap_nom` it is not specified default will be 0, just as for `tap_nom`

Reply:
Oh my bad. I guess I got confused by the double `?`

Reply:
we can also initialize `tap_nom` before `tap_pos` and then obtain `tap_pos` from `tap_nom`, e.g.

```cpp
tap_nom_{transformer_input.tap_nom == na_IntS ? IntS{0} : transformer_input.tap_nom},
tap_pos_{transformer_input.tap_pos == na_IntS ? tap_nom_ : transformer_input.tap_pos}
```

note that to stop (especially Unix) compilers from complaining, the order of the attributes at the declaration needs to be changed as well (`-Wreorder`)

Reply:
I think it should be fine like this. The documentation mentions how this initialization is done; if things are unclear